### PR TITLE
Deploy checkpointer independently from api-server

### LINKED ISSUE #161: checkpointer should be deployed independently from api-server
This can make upgrades of both checkpointer and api-server at the same time somewhat fragile. We should deploy this as a separate component so it can be upgraded independently.

### PR DESCRIPTION
**Pull request by kenan435**
Checkpointer deployed independently from api-server. Deploying it as a separate component so it can be upgraded independently.

fixes #161

### CODE REVIEW

File: pkg/asset/internal/templates.go [hunk 1]
```diff
@@ -168,13 +161,41 @@ spec:
...
           secretName: kube-apiserver
 `)
+	CheckpointerTemplate = []byte(`apiVersion: "extensions/v1beta1"
+kind: DaemonSet
+metadata:
+  name: kube-api-checkpointer
+  namespace: kube-system
+  labels:
+    k8s-app: kube-api-checkpointer
+    version: v1.0.0
```
Comment (aaronlevy) [inline pkg/asset/internal/templates.go]: Can you drop this version label

File: pkg/asset/internal/templates.go [hunk 2]
```diff
@@ -168,13 +161,41 @@ spec:
...
+spec:
+  template:
+    metadata:
+      labels:
+        k8s-app: kube-api-checkpointer
+        version: v1.0.0
```
Comment (aaronlevy) [inline pkg/asset/internal/templates.go]: Drop version label

File: pkg/asset/internal/templates.go [hunk 3]
```diff
@@ -168,13 +161,41 @@ spec:
...
-        hostPath:
-          path: /etc/kubernetes/manifests
       - name: secrets
         secret:
           secretName: kube-apiserver
 `)
+	CheckpointerTemplate = []byte(`apiVersion: "extensions/v1beta1"
+kind: DaemonSet
+metadata:
+  name: kube-api-checkpointer
```
Comment (aaronlevy) [inline pkg/asset/internal/templates.go]: We may want to name this `checkpoint-installer` to match what it is actually doing. /cc @derekparker

**Comment by aaronlevy** [comment]
Couple minor comments. Can you also be sure to sign the CLA so that check passes. And also will wait on a review from @derekparker

**Comment by kenan435** [comment]
Code review findings fixed. Look like the cla check passed?

**Pull request merged**